server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/TraceServiceElasticsearch.java

```diff
@@ -132,14 +134,69 @@ public Trace getFragment(String tenantId, String id) {
         return ret;
     }
 
+    protected List<Trace> getFragmentsForTraceId(String tenantId, String traceId) {
+        List<Trace> ret = new ArrayList<Trace>();
+
+        String index = client.getIndex(tenantId);
+
+        try {
+            RefreshRequestBuilder refreshRequestBuilder = client.getClient().admin().indices().prepareRefresh(index);
+            client.getClient().admin().indices().refresh(refreshRequestBuilder.request()).actionGet();
+
+            BoolQueryBuilder query = QueryBuilders.boolQuery()
+                    .must(QueryBuilders.termQuery("traceId", traceId));
+
+            SearchRequestBuilder request = client.getClient().prepareSearch(index)
+                    .setTypes(TRACE_TYPE)
+                    .setSearchType(SearchType.DFS_QUERY_THEN_FETCH)
+                    .setQuery(query)
+                    .addSort(ElasticsearchUtil.TIMESTAMP_FIELD, SortOrder.ASC);
```

Comment:
would be better to show results in descending order? I think that in most cases you are interested in the last/current state of the system.